src/Entities/Player.h

```diff
@@ -578,6 +578,9 @@ class cPlayer :
 	Otherwise it returns the dig speed (float GetDigSpeed(BLOCKTYPE a_Block)) divided by the block hardness (cBlockInfo::GetHardness(BLOCKTYPE a_Block)) divided by 30 if the player can harvest the block and divided by 100 if he can't. */
 	float GetPlayerRelativeBlockHardness(BLOCKTYPE a_Block);
 
+	void AddKnownItem(int ItemID);
+	std::map<int, bool> KnownItems;
```

Comment:
Use `std::set` instead.
Don't mix function definitions and member variables together. The general in-class order is: types, variables, functions.